tests/cli/preprocess_script_tests.py

```diff
@@ -0,0 +1,171 @@
+                        pos_idx=p,
+                        ext='.png',
+                    )
+                    cv2.imwrite(os.path.join(self.temp_path, im_name),
+                                self.im + c * 100)
+                    self.frames_meta = self.frames_meta.append(
+                        aux_utils.parse_idx_from_name(im_name),
+                        ignore_index=True,
+                    )
+        # Write metadata
+        self.frames_meta.to_csv(
+            os.path.join(self.temp_path, self.meta_name),
+            sep=',',
+        )
+        self.pp_config = {
+            'output_dir': self.output_dir,
+            'input_dir': self.temp_path,
+            'channel_ids': [0, 1, 3],
+            'resample_scale': 0.2,
+            'num_workers': 4,
+            'correct_flat_field': True,
+            'resize': {'scale_factor': 2,
+                        'resize_3d': False},
+            'masks': {'channels': [3],
+                      'str_elem_radius': 3},
+            'tile': {'tile_size': [10, 10],
+                     'step_size': [10, 10],
+                     'depths': [1, 1, 1],
+                     'mask_depth': 1,
+                     'image_format': 'zyx'},
+        }
+        self.base_config = {
+            'input_dir': self.temp_path,
+            'output_dir': self.output_dir,
+            'slice_ids': -1,
+            'time_ids': -1,
+            'pos_ids': -1,
+            'channel_ids': self.pp_config['channel_ids'],
+            'uniform_struct': True,
+            'int2strlen': 3,
+            'num_workers': 4,
+        }
+
+    def tearDown(self):
+        """
+        Tear down temporary folder and file structure
+        """
+        TempDirectory.cleanup_all()
+        nose.tools.assert_equal(os.path.isdir(self.temp_path), False)
+
+    def test_pre_process(self):
+        out_config, runtime = pp.pre_process(self.pp_config, self.base_config)
+
+        self.assertIsInstance(runtime, np.float)
+        self.assertEqual(
+            out_config['resize']['resize_dir'],
+            os.path.join(self.output_dir, 'resized_images')
+        )
+        self.assertEqual(
+            out_config['masks']['mask_dir'],
+            os.path.join(self.output_dir, 'mask_channels_3')
+        )
+        self.assertEqual(out_config['masks']['mask_out_channel'], 4)
+        self.assertEqual(
+            out_config['tile']['tile_dir'],
+            os.path.join(self.output_dir, 'tiles_10-10_step_10-10'),
+        )
+
+    def test_pre_process_resize3d(self):
+        cur_config = self.pp_config
+        cur_config['resize']['scale_factor'] = [2, 1, 1]
+        cur_config['resize']['resize_3d'] = True
+        out_config, runtime = pp.pre_process(cur_config, self.base_config)
+
+        self.assertIsInstance(runtime, np.float)
+        self.assertEqual(
+            out_config['resize']['resize_dir'],
+            os.path.join(self.output_dir, 'resized_images')
+        )
```

Comment:
could we also add an assert for input_dir and slice_ids please as they get modified in-place?https://github.com/czbiohub/microDL/blob/b0731eb724b30e013b6baf5fdb91f988cd1701eb/micro_dl/cli/preprocess_script.py#L284

Reply:
req_params_dict isn't an output of the preprocess script, onpy pp_config, and input dir doesn't change there...

Reply:
Same thing for slice_ids it seems... We could consolidate into one dict if you think that's wise? It might make sense to save these things if running preprocessing again.

Reply:
base_config is a subset of pp_config. In case of resize, we change these fields in base_config only and not in pp_config. The input_dir gets stored in pp_config['resize']['resize_dir'] = resize_dir but the modified slice_ids are not saved anywhere. Could we add them immediately after this: L 286: pp_config['input_dir'] = resize_dir & pp_config['slice_ids'] = slice_ids

However, we also lose the original config, that we started with a different input dir... we could add the slice_ids also under resize: pp_config['resize']['slice_ids'] = slice_ids. Either way works for me, whatever u think is best!